### DISCUSSION TITLE
405 Method not allowed error for my delete and patch requests but post and get requests work fine

### DISCUSSION THREAD

**Discussion by Benbellah-Owino** [tokio-rs/axum#2417] [Q&A] [answered]
### Summary

My get and post request are working fine but my patch and delete won't work.
Here is he routes.rs code
use crate::handlers::{create_todo, delete_todo, get_todo, update_todo};
use axum::routing::{post,self };
use axum::routing::{delete, patch};
use axum::Router;
use crate::db:: Db;

pub fn routes(db: Db) -> Router{
    
    Router::new()
        .route("/todo", post(create_todo).get(get_todo))
        .route("/todo/:id", delete(delete_todo).patch(update_todo))
        .with_state(db)
}
 For full code here is the project repository:->   https://github.com/Benbellah-Owino/axum_todo/tree/main

### axum version

0.7.1

**Comment by davidpdrsn** [comment, marked as answer]
There's gotta be something else wrong because that does work. I just tested this

```rust
use axum::{routing::*, Router};

#[tokio::main]
async fn main() {
    let app = Router::new()
        .route("/todo", post(|| async {}).get(|| async {}))
        .route("/todo/:id", delete(|| async {}).patch(|| async {}));

    let listener = tokio::net::TcpListener::bind("127.0.0.1:3000")
        .await
        .unwrap();
    axum::serve(listener, app).await.unwrap();
}
```

```
❯ curl -i localhost:3000/todo -X GET
HTTP/1.1 200 OK
content-length: 0
date: Mon, 11 Dec 2023 10:28:31 GMT


❯ curl -i localhost:3000/todo -X POST
HTTP/1.1 200 OK
content-length: 0
date: Mon, 11 Dec 2023 10:28:37 GMT


❯ curl -i localhost:3000/todo/1 -X DELETE
HTTP/1.1 200 OK
content-length: 0
date: Mon, 11 Dec 2023 10:28:42 GMT


❯ curl -i localhost:3000/todo/1 -X PATCH
HTTP/1.1 200 OK
content-length: 0
date: Mon, 11 Dec 2023 10:28:45 GMT




```

**Reply by Benbellah-Owino** [reply]
Ohh, ok. It still doesn't work for me but I think it's my error. Thanks for your time
